contracts/truefi2/PoolFactory.sol

```diff
@@ -234,6 +234,15 @@ contract PoolFactory is IPoolFactory, UpgradeableClaimable {
         emit BorrowerWhitelistStatusChanged(borrower, status);
     }
 
+    function isSupportedPool(ITrueFiPool2 _pool) external override view returns (bool) {
+        for (uint256 i = 0; i < supportedPools.length; i++) {
+            if (supportedPools[i] == _pool) {
```

Comment:
Let's store `supportedPools.length` in a local variable before the loop, so it doesn't query storage variable over each iteration.

Reply:
I don't think it affects gas cost in any way

Reply:
It does, but not as much as I thought. For 200 pools in `supportedPools` I'm getting around 20_000 gas saving (overall approx. 100 gas usage reduction per pool).

Reply:
I think checking array.length inside the for loop is fine. If there's not much gas savings, then this would be a premature optimization at the cost of a minor drop in readability. I could go either way, but I believe the rest of our repo checks array.length in the loop so for consistency we should do so here as well.

Thanks for noticing this and benchmarking it though!